### PR TITLE
Implement DAS and query for app specific data

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

Adds DAS and fetching optimint blocks from the data availability layer. 

closes: #5 #6

### CODE REVIEW

**Comment by Wondertan** [comment]
@evan-forbes, a small note that you can already implement querying of a specific namespace and checking if DAH is available as [#170](https://github.com/celestiaorg/celestia-node/pull/170) and [#171](https://github.com/celestiaorg/celestia-node/pull/171) are merged.

File: cmd/dalc/main.go
Comment (liamsi) [inline cmd/dalc/main.go:13]: Not sure if you chatted with @marbar3778  about this but wouldn't it make sense to merge https://github.com/celestiaorg/celestia-app/pull/177 first and the approach taken there instead of copy&pasting the cosmoscmd code?

Comment (liamsi) [reply cmd/dalc/main.go:13]: I see! Thanks for the clarification. I was not sure. It's OK to push this to a follow-up PR if others are blocked on this PR to land btw.

File: .github/workflows/test.yml
Comment (evan-forbes) [inline .github/workflows/test.yml:91]: deleted this cause it keeps breaking, and while it's a nice to have, I wouldn't define it as a must have

File: config/config.go
Comment (evan-forbes) [inline config/config.go:106]: this is a simple name change, but hopefully it helps differentiate this address from the rest api address

File: server/height_map.go
Comment (evan-forbes) [inline server/height_map.go:16]: @tzdybal perhaps keeping track of optimint block height: celestia block height, belongs in optimint. I didn't want to add yet another KV database, but we will have to do something in the long term, as using a simple map and saving it as a json will obviously not be ideal for a blockchain with a few million blocks.

File: server/server.go
Comment (evan-forbes) [inline server/server.go:154]: this still needs to be refactored to using celestia-node's header service, but there are already too many changes in this PR imho

File: server/server.go
Comment (evan-forbes) [inline server/server.go:186]: same comment as above, this will have to be changed when we refactor to actually use celestia-node as intented

File: server/test/server_integration_test.go
Comment (evan-forbes) [inline server/test/server_integration_test.go:29]: skipping this test, as it requires running a celestia-app network, connecting a celestia-node full node, and then properly configuring the dalc's celestia-node light as well. It can however be used to test things locally.

File: server/test/single-node.sh
Comment (evan-forbes) [inline server/test/single-node.sh:34]: this is the script that I was using with the integration test, we don't have to include it though.

**Review by evan-forbes — COMMENTED** [review]
Making this ready for review even though it still needs a refactor to properly utilize celestia-node, as it already contains too many changes. Instead of properly utilizing celestia-node, this version of the DALC is still downloading the entire celestia block to create the Data availability header by hand. 

I'll have an issue up before this is merged describing the changes that need to be made to

File: cmd/dalc/main.go
Comment (adlerjohn) [inline cmd/dalc/main.go:103]: This doesn't look cross-platform. Should `./celestia-light` also not be a constant?

File: server/height_map.go
Comment (tzdybal) [reply server/height_map.go:16]: I think keeping track is one of DALC functions, and Optimint should be agnostic. Also, I don't see any reason not to use KV store - it's operation won't be resource-consuming anyways. 

This implementation is going to work only for aggregator node (to be super specific: DALC instance that will be used by aggregator, will also be able to serve optimint block data to other nodes). I described in #24 results of our previous conversations about locating blocks in celestia chain.

IMHO we should merge as is and create follow-up issue.

File: config/config.go
Comment (evan-forbes) [reply config/config.go]: good ideas

added an issue https://github.com/celestiaorg/dalc/issues/38

also pushed some changes to use file path https://github.com/celestiaorg/dalc/pull/17/commits/4f932524bebee6693b602d8b6201ed80a80ac81b

File: config/config.go
Comment (evan-forbes) [reply config/config.go]: :+1: https://github.com/celestiaorg/dalc/pull/17/commits/4f932524bebee6693b602d8b6201ed80a80ac81b

File: cmd/dalc/main.go
Comment (evan-forbes) [reply cmd/dalc/main.go]: :+1: https://github.com/celestiaorg/dalc/pull/17/commits/65e799aa91b8bf413ffa2569a99526f33dc920c8

File: server/height_map.go
Comment (evan-forbes) [reply server/height_map.go]: :+1: https://github.com/celestiaorg/dalc/pull/17/commits/bd29e1daad6a23e8e99bde9464934859a1bc3d42

File: server/height_map.go
Comment (evan-forbes) [reply server/height_map.go:18]: :+1: follow-up https://github.com/celestiaorg/dalc/issues/39

File: server/server.go
Comment (evan-forbes) [reply server/server.go:154]: https://github.com/celestiaorg/dalc/issues/40

File: server/server.go
Comment (evan-forbes) [reply server/server.go:186]: https://github.com/celestiaorg/dalc/issues/40

File: config/config.go
Comment (adlerjohn) [inline config/config.go:15]: Celesti?
```suggestion
	CelestiaNodeHome = ".celestia-light"
```

Comment (evan-forbes) [reply config/config.go:15]: :man_facepalming:

**Review by tzdybal — APPROVED** [review]
As follow-up issues are created, I think we should move forward with this PR.

**Pull request merged**
